packages/backup-restore/src/commands/projects.ts

```diff
@@ -0,0 +1,107 @@
+import fs from "fs/promises";
+import path from "path";
+
+import { exists } from "../lib/helpers";
+import { configstore, ConfigOptions } from "../config/configstore";
+import { paths } from "../config/paths";
+
+import { IServiceAccountJson } from "../lib/firebase";
+import { FsErrors } from "../lib/types";
+
+/**
+ * List project IDs of available firebase credentials
+ */
+export function listProjectCredentials(): void {
+  const projects = configstore.get(ConfigOptions.Projects);
+  console.log(projects);
+}
+
+// TODO: following two functions could be fragmented and tested better
+/**
+ * Add firebase project credentials to XDG-specific AppData location
+ */
+export async function addProjectCredentials(filePath: string): Promise<void> {
+  // Resolve path to file and check it exists
+  const currentDir = process.cwd();
+  const resolvedPath = path.resolve(currentDir, filePath);
+
+  try {
+    const fileExists = await exists(resolvedPath);
```

Comment:
The docs for [`fs.access`](https://nodejs.org/api/fs.html#fsaccesspath-mode-callback) (used in the `exists` function) say:

>Do not use fs.access() to check for the accessibility of a file before calling fs.open(), fs.readFile(), or fs.writeFile(). Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

Wouldn't it be preferrable to let `fs.readFile` error out in case of a problem?